resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -397,6 +407,64 @@ private[spark] class KubernetesClusterSchedulerBackend(
     }
   }
 
+  /*
+ * KubernetesAllocator class watches all the executor pods associated with
+ * this SparkJob and creates new executors when it is appropriate.
+ */
+  private[spark] class KubernetesAllocator(
```

Comment:
Actually maybe it doesn't belong in a separate class at all - the reference to a `KubernetesClusterSchedulerBackend` in this class makes me think this logic actually belongs in the scheduler backend itself.